stubs/docker/docker/auth.pyi

```diff
@@ -1,40 +1,51 @@
-from _typeshed import Incomplete
+from _typeshed import FileDescriptorOrPath, Incomplete, ReadableBuffer
+from logging import Logger
+from typing import Final
 
-INDEX_NAME: str
-INDEX_URL: Incomplete
-TOKEN_USERNAME: str
-log: Incomplete
+INDEX_NAME: Final[str]
+INDEX_URL: Final[str]
+TOKEN_USERNAME: Final[str]
+log: Logger
 
-def resolve_repository_name(repo_name): ...
-def resolve_index_name(index_name): ...
-def get_config_header(client, registry): ...
-def split_repo_name(repo_name): ...
-def get_credential_store(authconfig, registry): ...
+def resolve_repository_name(repo_name: str) -> tuple[str, str]: ...
+def resolve_index_name(index_name: str) -> str: ...
+def get_config_header(client, registry) -> bytes | None: ...
+def split_repo_name(repo_name: str) -> tuple[str, str]: ...
+def get_credential_store(authconfig: AuthConfig | dict[str, Incomplete], registry: str | None): ...
 
 class AuthConfig(dict[str, Incomplete]):
-    def __init__(self, dct, credstore_env: Incomplete | None = None) -> None: ...
+    def __init__(self, dct: dict[str, Incomplete], credstore_env: Incomplete | None = None) -> None: ...
     @classmethod
-    def parse_auth(cls, entries, raise_on_error: bool = False): ...
+    def parse_auth(
+        cls, entries: dict[str, dict[Incomplete, Incomplete]], raise_on_error: bool = False
+    ) -> dict[str, Incomplete]: ...
     @classmethod
-    def load_config(cls, config_path, config_dict, credstore_env: Incomplete | None = None): ...
+    def load_config(
+        cls,
+        config_path: FileDescriptorOrPath | None,
+        config_dict: dict[str, Incomplete] | None,
```

Comment:
Might be implementation details, but any `MutableMapping` also works here.

```suggestion
        config_dict: MutableMapping[str, Incomplete] | None,
```

But you don't gain covariance like with a `Mapping`, and the param is explicitly named `*_dict`. So I think keeping this a `dict` is fine.

Reply:
I preferred to leave it as is